### PR TITLE
Add support for streaming hash digest calculation.

### PR DESCRIPTION
**Pull request by NorbertSzydlik**

### CODE REVIEW

File: src/gleam/crypto.gleam [hunk 1]
```diff
@@ -39,6 +39,23 @@ pub type HashAlgorithm {
...
+
+/// Starts hash algorithm for streaming
+@external(erlang, "crypto", "hash_update")
+@external(javascript, "../gleam_crypto_ffi.mjs", "hashUpdate")
+pub fn hash_update(state: HashState, data: BitArray) -> HashState
+
+/// Starts hash algorithm for streaming
+@external(erlang, "crypto", "hash_final")
+@external(javascript, "../gleam_crypto_ffi.mjs", "hashFinal")
+pub fn hash_final(state: HashState) -> BitArray
```
Comment (lpil) [inline src/gleam/crypto.gleam:57]: Not sure about these names. How about `Hasher`, `hasher`, `hash_chunk`, and `digest`?

**Review by lpil — COMMENTED** [review]
Very nice! Could you change the existing hash function to be implemented using this new API? And update its documentation to note the incremental API too.

We'll also need some examples in the doc comments showing how to use it, as it's a more complex API. Similar to how the stdlib shows examples. Thank you.

File: src/gleam/crypto.gleam (hunk 1 above)
Comment (NorbertSzydlik) [reply src/gleam/crypto.gleam:57]: Done

**Comment by NorbertSzydlik** [comment]
> Very nice! Could you change the existing hash function to be implemented using this new API? And update its documentation to note the incremental API too.
I've added comment in docs that `hash` function uses `hash_init`, `hash_update`, and `digest`

> We'll also need some examples in the doc comments showing how to use it, as it's a more complex API. Similar to how the stdlib shows examples. Thank you.
I have added one simple example, and one complex example using `actor`.

**Review by lpil — APPROVED** [review]
Thank you!

**Pull request merged**